core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -619,26 +618,19 @@ public FileSystemMasterView getFileSystemMasterView() {
   public List<AlluxioURI> checkConsistency(AlluxioURI path, CheckConsistencyOptions options)
       throws FileDoesNotExistException, InvalidPathException, IOException {
     List<AlluxioURI> inconsistentUris = new ArrayList<>();
-    Stack<LockedInodePath> pathsToCheck = new Stack<>();
-    try {
-      pathsToCheck.push(mInodeTree.lockInodePath(path, InodeTree.LockMode.READ));
-      while (!pathsToCheck.empty()) {
-        try (LockedInodePath currentPath = pathsToCheck.pop()) {
-          Inode currentInode = currentPath.getInode();
-          if (!checkConsistencyInternal(currentPath)) {
-            inconsistentUris.add(currentPath.getUri());
-          }
-          if (currentInode.isDirectory()) {
-            for (Inode child : ((InodeDirectory) currentInode).getChildren()) {
-              pathsToCheck
-                  .push(mInodeTree.lockFullInodePath(child.getId(), InodeTree.LockMode.READ));
-            }
-          }
+    try (LockedInodePath parent = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ);
+        InodeLockList children = mInodeTree.lockDescendants(parent, InodeTree.LockMode.READ)) {
+      HashMap<Long, AlluxioURI> directoryPaths = new HashMap<>();
+      directoryPaths.put(parent.getInode().getId(), parent.getUri());
+      for (Inode child : children.getInodes()) {
+        AlluxioURI parentPath = directoryPaths.get(child.getParentId());
+        AlluxioURI currentPath = parentPath.join(child.getName());
```

Comment:
I think you can use `InodeTree.getPath()` to get the URI for an already locked inode.